pkg/controller/environment.go

```diff
@@ -132,8 +132,8 @@ func (env *K8sEnvironment) Init(cont *AciController) error {
 			Services(service.ObjectMeta.Namespace).UpdateStatus(service)
 	}
 
-	cont.listServicesBySelector = func(selector string) (*v1.ServiceList, error) {
-		return kubeClient.CoreV1().Services("").List(metav1.ListOptions{LabelSelector: selector})
+	cont.listServicesBySelector = func(labelSelector string, fieldSelector string) (*v1.ServiceList, error) {
+		return kubeClient.CoreV1().Services("").List(metav1.ListOptions{LabelSelector: labelSelector, FieldSelector: fieldSelector})
```

Comment:
Can you list the services from local store right? do you need to make a client call here ? otherwise  changes looks good

Reply:
Done!